.github/workflows/test-and-deploy.yml

```
@@ -0,0 +1,105 @@
+      - uses: actions/checkout@v4
+      - name: Run unit tests
+        run: docker-compose -f actions-services.yml run --rm app ./run-tests.sh
+
+  apidocs:
+    name: Update api docs
+    if: github.ref_name == 'develop'
+    runs-on: ubuntu-latest
+    env:
+      S3_PATH: ${{ vars.S3_PATH }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Update api docs
+        run: docker-compose -f actions-services.yml run --rm docs ./build-push-api-docs.sh
+
+  build-and-deploy:
+    name: Build and Deploy
+    if: github.ref_name == 'main' || github.ref_name == 'develop'
+    needs:
+      - test
+    outputs:
+      image: ${{ steps.push-image.outputs.image_to_deploy }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup
+        id: setup
+        run: |
+          echo "timestamp=$(date +'%s')" >> $GITHUB_OUTPUT
+          case "$BRANCH_NAME" in
+          "main") echo "environment=prod" >> $GITHUB_OUTPUT ;;
+          *) echo "environment=stg" >> $GITHUB_OUTPUT ;;
+          esac
+
+      - name: Build Image
+        id: build-image
+        run: |
+          docker build -t $IMAGE_NAME:$BRANCH_NAME application
+          echo "image=$IMAGE_NAME:$BRANCH_NAME" >> $GITHUB_OUTPUT
+
+      - name: Install dfctl
+        run: curl https://repo.deepfactor.io/install-dfctl.sh | sh --
+
+      - name: Deepfactor Scan
+        env:
+          DF_RUN_TOKEN: ${{ secrets.DF_RUN_TOKEN }}
+        run: dfctl scan ${{ steps.build-image.outputs.image }}
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Login to AWS ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v2
+
+      - name: Push Images to AWS
+        id: push-image
+        env:
+          BASE_IMAGE_WITH_TAG: ${{ steps.build-image.outputs.image }}
+          REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          TAG: ${{ steps.setup.outputs.timestamp }}
+          ENVIRONMENT: ${{ steps.setup.outputs.environment }}
+        run: |
+          docker tag $BASE_IMAGE_WITH_TAG $REGISTRY/$IMAGE_NAME-$ENVIRONMENT:$TAG
+          docker push $REGISTRY/$IMAGE_NAME-$ENVIRONMENT:$TAG
+          docker tag $BASE_IMAGE_WITH_TAG $REGISTRY/$IMAGE_NAME-$ENVIRONMENT:$BRANCH_NAME
+          docker push $REGISTRY/$IMAGE_NAME-$ENVIRONMENT:$BRANCH_NAME
+          echo "image_to_deploy=$REGISTRY/$IMAGE_NAME-$ENVIRONMENT" >> $GITHUB_OUTPUT
+
+      - name: Install ecs-deploy
+        run: |
+          curl -o /usr/local/bin/ecs-deploy https://raw.githubusercontent.com/silinternational/ecs-deploy/master/ecs-deploy
+          chmod +x /usr/local/bin/ecs-deploy
+
+      - name: Deploy to Web
+        id: deploy-image
+        env:
+          AWS_DEFAULT_REGION: ${{ vars.AWS_REGION }}
```

Optional: The ecs-deploy script, for some unknown reason, defined `AWS_DEFAULT_REGION` but not `AWS_REGION`. Nevertheless, you can use `AWS_REGION` as it will just be passed into the `aws` CLI for its use.
```suggestion
          AWS_REGION: ${{ vars.AWS_REGION }}
```

---

The `AWS_REGION` is a workflow env var, I was debating sticking everything only where it was used, but didn't get around to optimizing yet.

---

I think the change I suggested would still work since it's the name of the variable set in the process where `ecs-deploy` runs.